### DISCUSSION TITLE
AlpineJS conditional classes on blade components

### DISCUSSION THREAD

**Discussion by ju5t** [alpinejs/alpine#722] [4. General]
Is it possible to add conditional classes to blade components? 

I opened an issue/question with Blade UI Kit as the class didn't get applied and I thought it may be related to the attribute parsing happening there, but it may be an issue in AlpineJS (?) or perhaps the components itself.

Ref: https://github.com/blade-ui-kit/blade-icons/issues/80

**Comment by MuzafferDede** [comment]
I think it is how blade behaves on `:` attributes. Maybe you should use `x-bind:class="{'class' : variable}"` since `:class` is just an alias of `x-bind:class`.

**Comment by ju5t** [comment]
Thanks! It works. Is there a specific reason why `x-bind` works and only using `:class` doesn't?

**Reply by MuzafferDede** [reply]
When using a BladeX component all attributes assigned with `:`  will be passed as variables to the underlying Blade view. So `:class="['somthing']"` will be pass to view as `$class = ['something']`. 

For future use cases do not use `@` and `:` shorthands for alpine on blade views. Instead use `x-bind:` and `x-on:`

**Reply by ju5t** [reply]
Got it. Thanks for very quick reply!
